metric_learn/constraints.py

```diff
@@ -59,18 +59,22 @@ def _pairs(self, num_constraints, same_label=True, max_iter=10):
     ab = np.array(list(ab)[:num_constraints], dtype=int)
     return self.known_label_idx[ab.T]
 
-  def chunks(self, num_chunks=100, chunk_size=2):
+  def chunks(self, num_chunks=100, chunk_size=2, random_state=np.random):
+    """
+    the random state object to be passed must be a numpy random seed
+    """
+    state = random_state
     chunks = -np.ones_like(self.known_label_idx, dtype=int)
     uniq, lookup = np.unique(self.known_labels, return_inverse=True)
     all_inds = [set(np.where(lookup==c)[0]) for c in xrange(len(uniq))]
     idx = 0
     while idx < num_chunks and all_inds:
-      c = random.randint(0, len(all_inds)-1)
+      c = np.random.randint(0, high=len(all_inds)-1)
```

Comment:
This is a good start, but we need to be using `random_state` here:

    c = random_state.randint(0, high=len(all_inds)-1)

(Same comment for all the other places where we're directly calling functions on `np.random` in this module.)